internal/storage/postgres/tree/get_test.go

```diff
@@ -0,0 +1,534 @@
+		r := NewTreeRepository(suite.Store, mappers)
+
+		// when
+		tree, err := r.GetByID(context.Background(), 1)
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, tree)
+		assert.NotNil(t, tree.TreeCluster)
+		assert.NotNil(t, tree.Sensor)
+		assertExpectedEqualToTree(t, tree, testTrees[0])
+	})
+	t.Run("should return error if tree not found", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+
+		// when
+		tree, err := r.GetByID(context.Background(), 99)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, tree)
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewTreeRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		tree, err := r.GetByID(ctx, 1)
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, tree)
+	})
+}
+func TestTreeRepository_GetTreesByIDs(t *testing.T) {
+	t.Run("should return trees successfully by IDs", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
+		r := NewTreeRepository(suite.Store, mappers)
+		ids := []int32{1, 2}
+		expectedTrees := testTrees[:2]
+
+		// when
+		trees, err := r.GetTreesByIDs(context.Background(), ids)
+
+		// then
+		assert.NoError(t, err)
+		assert.NotNil(t, trees)
+		assert.Len(t, trees, len(expectedTrees))
+		for i, tree := range trees {
+			assertExpectedEqualToTree(t, expectedTrees[i], tree)
+		}
+	})
+
+	t.Run("should return empty list if no trees are found", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		r := NewTreeRepository(suite.Store, mappers)
+		ids := []int32{99, 100}
+
+		// when
+		trees, err := r.GetTreesByIDs(context.Background(), ids)
+
+		// then
+		assert.NoError(t, err)
+		assert.Empty(t, trees)
+	})
+
+	t.Run("should return error when context is canceled", func(t *testing.T) {
+		// given
+		r := NewTreeRepository(suite.Store, mappers)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		// when
+		trees, err := r.GetTreesByIDs(ctx, []int32{1, 2})
+
+		// then
+		assert.Error(t, err)
+		assert.Nil(t, trees)
+	})
+}
+func TestTreeRepository_GetCenterPoint(t *testing.T) {
+	t.Run("should successfully calculate and parse the center point", func(t *testing.T) {
+		// given
+		suite.ResetDB(t)
+		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
+		r := NewTreeRepository(suite.Store, mappers)
+		ids := []int32{1, 2, 3}
+		centroids, err := suite.Store.CalculateGroupedCentroids(context.Background(), ids)
+		if err != nil {
+			t.Fatal(err)
+		}
+		g, errGeos := geos.NewGeomFromWKT(centroids)
+		if errGeos != nil {
+			t.Fatal(errGeos)
+		}
+
+		// when
+		lat, long, errRepo := r.GetCenterPoint(context.Background(), ids)
+
+		// then
+		assert.NoError(t, errRepo)
+		assert.Equal(t, g.X(), lat, "Latitude does not match")
+		assert.Equal(t, g.Y(), long, "Longitude does not match")
```

Comment:
Why is `"Longitude does not match"` returned on successful?

Reply:
this is the error message. 
Please take a look at this link, it might be helpful: https://pkg.go.dev/github.com/stretchr/testify/assert